Command with id ‘workbench.files.action.focusFilesExplorer’ is not registered

[6/1/2020 7:20:02 PM] [SUCCESS] Successfully generated Smart Contract Project
[6/1/2020 7:20:03 PM] [ERROR] Issue creating Smart Contract Project: Error: Command with id ‘workbench.files.action.focusFilesExplorer’ is not registered.

This command got introduced into Eclipse Theia recently, so it should appear in Eclipse Che and CodeReady Workspaces eventually: https://github.com/eclipse-theia/theia/pull/6921

We should just wrap it in a try/catch.

--- a/packages/blockchain-extension/extension/commands/createSmartContractProjectCommand.ts
+++ b/packages/blockchain-extension/extension/commands/createSmartContractProjectCommand.ts
@@ -151,7 +151,11 @@ export async function createSmartContractProject(): Promise<void> {
 
         // Open the returned folder in explorer, in a new window
         await UserInputUtil.openNewProject(openMethod, folderUri);
-        await vscode.commands.executeCommand('workbench.files.action.focusFilesExplorer');
+        try {
+            await vscode.commands.executeCommand('workbench.files.action.focusFilesExplorer');
+        } catch (error) {
+            // This command is not supported on Eclipse Che, so just ignore the error.
+        }
     } catch (error) {
         outputAdapter.log(LogType.ERROR, `Issue creating${privateOrDefault}Smart Contract Project: ${error.message}`, `Issue creating${privateOrDefault}Smart Contract Project: ${error.toString()}`);
         return;

--- a/packages/blockchain-extension/test/commands/createSmartContractProjectCommand.test.ts
+++ b/packages/blockchain-extension/test/commands/createSmartContractProjectCommand.test.ts
@@ -303,6 +303,20 @@ describe('CreateSmartContractProjectCommand', () => {
         sendTelemetryEventStub.should.have.been.calledOnceWithExactly('createSmartContractProject', { contractLanguage: 'typescript', contractType: 'private' });
     });
 
+    it('should not report an error when focusFilesExplorer throws an error', async () => {
+        executeCommandStub.withArgs('workbench.files.action.focusFilesExplorer').rejects(new Error('unsupported'));
+        showQuickPickItemStub.resolves({label: UserInputUtil.GENERATE_PD_CONTRACT, description: UserInputUtil.GENERATE_PD_CONTRACT_DESCRIPTION, data: 'private'});
+        showInputBoxStub.onFirstCall().resolves('Org1MSP');
+        showLanguagesQuickPickStub.resolves({ label: 'TypeScript', type: LanguageType.CONTRACT });
+        showInputBoxStub.onSecondCall().resolves('MyPrivateAsset');
+        showFolderOptionsStub.resolves(UserInputUtil.OPEN_IN_NEW_WINDOW);
+        browseStub.resolves(uri);
+
+        await vscode.commands.executeCommand(ExtensionCommands.CREATE_SMART_CONTRACT_PROJECT);
+        logSpy.should.have.been.calledWith(LogType.SUCCESS, 'Successfully generated Private Data Smart Contract Project');
+        logSpy.should.not.have.been.calledWithExactly(LogType.ERROR, sinon.match.string, sinon.match.string);
+    });
+
     it('should not do anything if the user cancels when asked for the mspID', async () => {
         showQuickPickItemStub.resolves({label: UserInputUtil.GENERATE_PD_CONTRACT, description: UserInputUtil.GENERATE_PD_CONTRACT_DESCRIPTION, data: 'private'});
         showInputBoxStub.onFirstCall().resolves(undefined);

